design/proposals/dbt-delete-merge-schema-branch.md

```diff
@@ -0,0 +1,45 @@
+# Add DBT delete and merge branch-schema operations
+
+## Background
+
+When using dbt alongside lakeFS, the first step is to create a branch to work on and then create a dbt schema dedicated for that branch (that might have the same name of the branch [by default] or a custom one). #2988 added an option to use `lakectl dbt create-branch-schema --create-branch` to create the dedicated branch and schema at once.
+
+After running dbt over that branch and conducting tests and validations we would want to either: 
+- delete the branch and the schema associated with it
+- merge them into master/main/production/other branch
+- both.
+
+## Requirements
+
+The users that use the dbt command are, well, dbt users. They will interact with the `lakectl` tool in the context of the dbt command (and some basic commands such as `branch`, `commit` and `merge`).
+Adding the `delete-branch-schema` and `merge-branch-schema` subcommands will make it easier for them to integrate lakeFS into their framework.
+
+### Goals 
+- Delete schema, tables, partitions and associated metadata from the metastore using a single command.
+- Merge schema, tables, partitions and associated metadata from the metastore using a single command.
+
+### Non-goals
+
+- Deleting a branch before/after deleting an associated schema.
+- Merging a branch before/after deleting an associated schema.
+
+## Suggestions
+
+- `delete-branch-schema` will allow deleting a schema based on a branch name.
+- `merge-branch-schema` will allow merging a schema and all dbt model underneath it to a destination schema by branch names.
```

Comment:
```suggestion
- `merge-branch-schema` will allow merging a schema and all dbt models underneath it to a destination schema by branch names.
```

What does this mean to merge the dbt models under a schema? isn't a dbt model a piece of code?

Reply:
Yes you're correct. I used it interchangeably with the materialization of the model instead (a table or a view).
I'll change that.